src/formatter/formatter.vala

```diff
@@ -0,0 +1,278 @@
+using Gee;
+
+class Vls.Formatter : Object {
```

Comment:
I don't think we need to have the formatter as a class. Instead we can have static functions inside a namespace `Vls.Formatter`. This is how other code help routines are implemented (see `SymbolReferences` for example).